test cases/common/197 function attributes/meson.build

```diff
@@ -117,6 +117,12 @@ if ['gcc', 'intel'].contains(c.get_id())
   endif
```

Comment:
This if statement tests something that is impossible. This whole test needs a little bit of refactoring, I think.

Reply:
Yeah, there's a bunch of things in here that should be refactored it looks like